catalog/controller/extension/payment/razorpay.php

```diff
@@ -842,4 +867,97 @@ public function update()
             return  $this->response->redirect($this->url->link('extension/payment/razorpay/info', 'subscription_id=' . $postData['subscriptionId'], true));
         }
     }
+
+    protected function processSubscriptionAction($data)
```

Comment:
Give proper name here not getting what this function does based on the function name